hibernate-reactive-core/src/main/java/org/hibernate/reactive/engine/impl/ReactiveCollectionRecreateAction.java

```diff
@@ -0,0 +1,110 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive.engine.impl;
+
+import java.io.Serializable;
+import java.util.concurrent.CompletionStage;
+
+import org.hibernate.HibernateException;
+import org.hibernate.action.internal.CollectionAction;
+import org.hibernate.action.spi.AfterTransactionCompletionProcess;
+import org.hibernate.action.spi.BeforeTransactionCompletionProcess;
+import org.hibernate.collection.spi.PersistentCollection;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.event.service.spi.EventListenerGroup;
+import org.hibernate.event.spi.EventType;
+import org.hibernate.event.spi.PostCollectionRecreateEvent;
+import org.hibernate.event.spi.PostCollectionRecreateEventListener;
+import org.hibernate.event.spi.PreCollectionRecreateEvent;
+import org.hibernate.event.spi.PreCollectionRecreateEventListener;
+import org.hibernate.persister.collection.CollectionPersister;
+import org.hibernate.reactive.engine.ReactiveExecutable;
+import org.hibernate.reactive.persister.collection.impl.ReactiveCollectionPersister;
+
+
+
+//FIXME: Not sure if we should extend CollectionAction. I guess we can leave it for now
+// and see later if we need to change the methods in the superclass
+public class ReactiveCollectionRecreateAction extends CollectionAction implements ReactiveExecutable {
+
+	public ReactiveCollectionRecreateAction(
+			final PersistentCollection collection,
+			final CollectionPersister persister,
+			final Serializable key,
+			final SharedSessionContractImplementor session) {
+		super( persister, collection, key, session );
+	}
+
+	@Override
+	public CompletionStage<Void> reactiveExecute() {
+		final ReactiveCollectionPersister persister = (ReactiveCollectionPersister)getPersister();
+		final SharedSessionContractImplementor session = getSession();
+		final Serializable key = getKey();
+
+		final PersistentCollection collection = getCollection();
+
+		preRecreate();
+
+		return persister.recreateReactive( collection, key, session ).thenAccept( v -> {
+			session.getPersistenceContextInternal().getCollectionEntry( collection ).afterAction( collection );
+			evict();
+			postRecreate();
+		} );
+	}
+
+	@Override
+	public int compareTo(Object o) {
+		return 0;
+	}
+
+	@Override
+	public Serializable[] getPropertySpaces() {
+		return new Serializable[0];
+	}
+
+	@Override
+	public AfterTransactionCompletionProcess getAfterTransactionCompletionProcess() {
+		return null;
+	}
+
+	@Override
+	public BeforeTransactionCompletionProcess getBeforeTransactionCompletionProcess() {
+		return null;
```

Comment:
Are these methods correctly implemented?

Reply:
A bit of comments here would help

Reply:
I think the overrides in the 3 Actions were redundant... left over from working from the CollectionAction.  Cleaning them up.